add march_input_device tutorial and overview (#51)
project-march/tutorials: doc/march_packages/march_input_device.rst

march_input_device
==================

Overview
--------

> The march_input_device is the software running on the |m4| exoskeleton input device. The input device is used by the

Author: ```suggestion
The march_input_device is the software running on the |m4| exoskeleton input device. The input device is used
```

> to give input to the software running on the |m4| exoskeleton.

Reviewer: By the what?

Author: fixed

Communication
-------------
The communication between the input device and the |m4| exoskeleton is done with `rosserial <http://wiki.ros.org/rosserial>`_.
This can be run wired via usb cable or wireless via wifi connection.

> Sending messages

Reviewer: Link to the message files

Author: done

^^^^^^^^^^^^^^^^
**Gait instruction:** This is a instruction to start the sent gait right now, if this is possible.

> **Stop:** This message can stop a looping gait(such as walk). After receiving the stop message the |m4| exoskeleton will go to the standing pose.

Reviewer: ```suggestion
**Stop:** This message can stop a repeating gait (such as walk). After receiving the stop message the exoskeleton will go to the standing pose.
```

.. todo:: (Tim) link this to the march_safety documentation.

> **Stay alive:** Every cycle a stay alive message is send. This way *march_safety* can detect when the input device loses connection.

Author: ```suggestion
**Stay alive:** Every loop a stay alive message is send. This way *march_safety* can detect when the input device loses connection.
```

> **Stay alive:** Every cycle a stay alive message is send. This way *march_safety* can detect when the input device loses connection.

Reviewer: What cycle?

Receiving messages
^^^^^^^^^^^^^^^^^^

> **Gait instruction response:** This message indicate that the last gait instruction is handled. The value represent successful handled or rejected.

Author: ```suggestion
**Gait instruction response:** This message indicates that the last gait instruction is received. The value tells you if the instruction is successful handled or rejected.
```

Author: Is this better?

> **Gait instruction response:** This message indicate that the last gait instruction is handled. The value represent successful handled or rejected.

Reviewer: indicates*
represents*

What does handled mean?

Tutorials
---------

How to run wired
^^^^^^^^^^^^^^^^

.. todo:: (Electro) Document how to change the input device to wired mode

Upload the code on the input device
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Skip this section if the correct code for running wired is already on the input device. More detailed information on how
to run and upload the code see the :input-device:`ReadMe <>`.

- Comment :code:`#define USE_WIRELESS` in the main.cpp.
- Upload to input device.

Start the software
~~~~~~~~~~~~~~~~~~
- Type in a terminal:

.. code::

>     roscore

Reviewer: Why start a core separately?

- Type in a terminal:

.. code::

    roslaunch march_launch serial_connection.launch tcp:=false

- Connect the input device via usb cable, this powers and launches the input device automatically.

How to run Wireless
^^^^^^^^^^^^^^^^^^^

.. todo:: (Electro) Document how to change the input device to wireless mode

> .. note:: Make sure the pc that needs to receive the messages and the input device are on the same hotspot.

Reviewer: ```suggestion
.. note:: Make sure the pc that needs to receive the messages and the input device are on the same network.
```

With a router it's technically not a hotspot. Message is clear though

Upload the code on the input device
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Skip this section if the correct code for running wired is already on the input device. More detailed information on how
to run and upload the code see the :input-device:`ReadMe <>`.

> .. note:: If you want to change the hotspot name and password that the input device uses: Change **ssid** and **password** in the *WirelessConnection.h*

Author: ```suggestion
.. note:: If you want to change the network name and password that the input device uses: Change **ssid** and **password** in the *WirelessConnection.h*
```

> .. note:: If you want to change the hotspot name and password that the input device uses: Change **ssid** and **password** in the *WirelessConnection.h*

Reviewer: Once again not hotspot
